ion-schema/src/main/kotlin/com/amazon/ionschema/internal/constraint/Occurs.kt

```diff
@@ -92,36 +83,12 @@ internal class Occurs(
     private val typeReference: () -> TypeInternal
 
     init {
-        var occurs: IonValue? = null
-        range =
-            if (ion is IonStruct && !ion.isNullValue) {
-                occurs = ion["occurs"]
-                if (occurs != null) {
-                    toRange(occurs)
-                } else {
-                    defaultRange
-                }
-            } else {
-                defaultRange
-            }
+        val occurs: IonValue? = (ion as? IonStruct)?.takeIf { !ion.isNullValue }?.get("occurs")
 
-        val tmpIon = if (ion is IonStruct && occurs != null) {
-            ion.cloneAndRemove("occurs")
```

Comment:
🗺️ Removing this line is the only change in this file that is part of the actual fix for prohibiting `nullable` and `$null_or` on variably occurring type definitions. All of the rest of the changes in this file are just cleaning up some overly complicated code.